src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -67,6 +67,11 @@ internal partial class Http2Connection : IHttp2StreamLifetimeHandler, IHttpHeade
         private int _gracefulCloseInitiator;
         private int _isClosed;
 
+        private Http2StreamStack _streamPool;
+
+        internal const int InitialStreamPoolSize = 20;
+        internal const int MaxStreamPoolSize = 200;
```

Comment:
I expect this is still higher than it needs to be. Can you try benchmarks at a few different values like 20, 50, and 100? I anticipate rapidly diminishing returns.